packages/react-components/react-table/Spec.md

```diff
@@ -2,62 +2,373 @@
+more mature into development.
+
+```ts
+interface TableProps {
+  /**
+   * Determines sizes of child components
+   */
+  size: 'small' | 'smaller' | 'large';
+  /**
+   * Whether the columns are sortable
+   */
+  sortable?: boolean;
+
+  /**
+   * Determines if/how the table can be navigated by keyboard
+   *
+   * @default none
+   */
+  navigationMode?: 'none' | 'cell' | 'row' | 'composite';
+}
+
+interface TableHeaderCell {
+  /**
+   * Whether the column is sortable overrides the top level table prop
+   */
+  sortable: boolean;
+  /**
+   * How the column data is sorted
+   */
+  sortDirection: 'ascending' | 'descending';
+}
+
+/**
+ * Layout and design sugar for the primary column of a table
+ */
+interface TablePrimaryCell {
+  secondaryContent: Slot<'span'>;
+  icon: Slot<'span'>;
+}
+```
 
 ## Structure
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+### Basic
 
-## Migration
+```tsx
+<Table>
+  <TableHeader>
+    <TableRow>
+      <TableCell> </TableCell>
+    <TableRow>
+  </TableHeader>
+
+  <TableBody>
+    <TableRow>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
+
+```html
+<div role="table">
+  <div role="rowgroup">
+    <div role="row">
+      <div role="columnheader"></div>
+    </div>
+
+    <div role="row"></div>
+  </div>
+
+  <div role="rowgroup">
+    <div role="row">
+      <div role="cell"></div>
+    </div>
+  </div>
+</div>
+```
+
+### Sortable
+
+```tsx
+<Table sortable>
+  <TableHeader>
+    <TableRow>
+      <TableHeaderCell sortDirection="ascending">Header</TableHeaderCell>
+    </TableRow>
+  </TableHeader>
+</Table>
+```
+
+```html
+<div role="table">
+  <div role="rowgroup">
+    <div role="row">
+      <div role="columnheader" aria-sort="ascending"><button>Header</button></div>
+    </div>
+  </div>
+</div>
+```
+
+### Primary column
+
+```tsx
+<Table>
+  <TableBody>
+    <TableRow>
+      <TablePrimaryCell secondaryContent="Secondary content" icon={<FileIcon />}>
+        Primary content
+      </TablePrimaryCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
+
+```html
+<div role="table">
+  <div role="rowgroup">
+    <div role="row">
+      <div role="cell">
+        <span aria-hidden="true">icon</span><span>Primary content</span> <span>Secondary content</span>
+      </div>
+    </div>
+  </div>
+</div>
+```
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+## Migration
 
-- _Migration from v8_
-- _Migration from v0_
+> ⚒️ This section is still being developed and will be completed once APIs for this component become more stable
 
 ## Behaviors
 
-_Explain how the component will behave in use, including:_
+### Sortable header cells
+
+Table header cells are only focusable when they are sortable. Focus when tabbing into the Table control should
+focus on the first sortable header, if any.
+
+### Navigation modes
```

Comment:
Does this component need to handle scrolling/overflow? For example, a style sets the height of the component to `200px` but there is `600px` of content.

Reply:
Maybe overflow isn't the right term but I'm thinking of cases where header rows and columns are fixed. Excel and Windows Explorer/macOS Finder are examples. This is probably more of a `DataGrid` feature rather than a `Table` one though I could see it being used for tables.

Some examples:

### Excel Online:
![excel_fixed_headers](https://user-images.githubusercontent.com/93940821/180039389-0419cc32-6b30-4b67-9b2a-4d7b688fe55b.gif)

### Windows File Explorer
![file_explorer](https://user-images.githubusercontent.com/93940821/180039417-3b6f022d-9919-4245-a760-b6e828496554.gif)

Reply:
The seems to be rather sticky header and columns. I could see how simple css could make headers and columns sticky, especially with an API where the user has access to much more of the DOM than `DetailsList` to apply overrides more easily.

I'll keep this in mind and after the first few iterations evaluate this scenario a bit more